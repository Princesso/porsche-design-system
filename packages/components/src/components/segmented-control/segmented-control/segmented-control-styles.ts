import { buildResponsiveStyles, getCss } from '../../../utils';
import { addImportantToEachRule, hostHiddenStyles } from '../../../styles';
import type { BreakpointCustomizable } from '../../../types';

const MIN_ITEM_WIDTH = 46;
const MAX_ITEM_WIDTH = 220;

export const getComponentCss = (maxItemWidth: number, columns: BreakpointCustomizable<number | 'auto'>): string => {
  const minWidth =
    (maxItemWidth > MAX_ITEM_WIDTH && MAX_ITEM_WIDTH) ||
    (maxItemWidth < MIN_ITEM_WIDTH && MIN_ITEM_WIDTH) ||
    maxItemWidth;

  return getCss({
    '@global': {
      ':host': addImportantToEachRule({
        display: 'grid',
        gridAutoRows: '1fr', // for equal height
        ...buildResponsiveStyles(columns, (col: number | 'auto') => ({
          gridTemplateColumns:
            col === 'auto' ? `repeat(auto-fit, ${minWidth}px)` : `repeat(${col}, minmax(${minWidth}px, 1fr))`,
        })),
        gap: '6px',
        ...hostHiddenStyles,
      }),
    },
  });
};
