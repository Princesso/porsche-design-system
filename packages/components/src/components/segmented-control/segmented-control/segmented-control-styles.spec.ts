import { getComponentCss } from './segmented-control-styles';

xdescribe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([[20], [80], [230]])(
    'should return correct css for maxWidth: %s',
    (...args) => {
      expect(getComponentCss(...args)).toMatchSnapshot();
    }
  );
});
