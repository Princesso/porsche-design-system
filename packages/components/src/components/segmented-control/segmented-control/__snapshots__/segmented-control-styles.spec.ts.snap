// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for maxWidth: 20 and columns: { base: 4, s: 3, m: 2, l: 'auto' } 1`] = `
":host {
  display: grid !important;
  grid-auto-rows: 1fr !important;
  grid-template-columns: repeat(4, 46px) !important;
  gap: 6px !important;
}
:host([hidden]) {
  display: none !important;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: repeat(3, 46px) !important;
  }
}

@media(min-width:1000px) {
  :host {
    grid-template-columns: repeat(2, 46px) !important;
  }
}

@media(min-width:1300px) {
  :host {
    grid-template-columns: repeat(auto-fit, minmax(46px, 1fr)) !important;
  }
}
"
`;

exports[`getComponentCss() should return correct css for maxWidth: 20 and columns: auto 1`] = `
":host {
  display: grid !important;
  grid-auto-rows: 1fr !important;
  grid-template-columns: repeat(auto-fit, minmax(46px, 1fr)) !important;
  gap: 6px !important;
}
:host([hidden]) {
  display: none !important;
}"
`;

exports[`getComponentCss() should return correct css for maxWidth: 80 and columns: 1 1`] = `
":host {
  display: grid !important;
  grid-auto-rows: 1fr !important;
  grid-template-columns: repeat(1, 80px) !important;
  gap: 6px !important;
}
:host([hidden]) {
  display: none !important;
}"
`;

exports[`getComponentCss() should return correct css for maxWidth: 230 and columns: 10 1`] = `
":host {
  display: grid !important;
  grid-auto-rows: 1fr !important;
  grid-template-columns: repeat(10, 220px) !important;
  gap: 6px !important;
}
:host([hidden]) {
  display: none !important;
}"
`;
